sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_crypto.c

```diff
@@ -136,30 +257,89 @@ void gnrc_lorawan_decrypt_join_accept(const uint8_t *key, uint8_t *pkt,
     }
 }
 
-void gnrc_lorawan_generate_session_keys(const uint8_t *app_nonce,
-                                        const uint8_t *dev_nonce,
-                                        const uint8_t *appkey, uint8_t *nwkskey,
-                                        uint8_t *appskey)
+#if IS_ACTIVE(CONFIG_LORAWAN_1_1)
```

Comment:
If you can join these two functions it would be also nice

Reply:
Can you explain to me why joining functions and using a struct as parameter that has different members depending on the LoRaWAN version used is a better style than just using two functions ? 
To me having 2 separate functions is way cleaner.

Reply:
+ Using two functions avoids having to initialize a struct that takes up stack space.

Reply:
> To me having 2 separate functions is way cleaner.

To be clear, I'm not arguing for using a struct with different elements rather than two functions. Of course two functions are better when the functions diverge. But in this case, LoRaWAN 1.0.x is still a subset of LoRaWAN 1.1.
Therefore, these function cannot be _that_ divergent (and they don't seem to diverge either).
If you think there are some snippets that don't make sense at all to merge, it is fine. But I think we should try to avoid having the same snippets spread on different functions if they perform almost the same task.